codex/blockexchange/engine/engine.nim

```diff
@@ -77,6 +78,7 @@ type
     pricing*: ?Pricing                            # Optional bandwidth pricing
     blockFetchTimeout*: Duration                  # Timeout for fetching blocks over the network
     discovery*: DiscoveryEngine
+    advertiser*: Advertiser
```

Comment:
The way I see it, discovery: DiscoveryEngine and advertiser: Advertiser are each other's counterparts. They even have similar structures, and they exist side-by-side. Discovery discovers what advertiser advertises. (mind = blown) I am quite sure that these field are only marked as public because of tests.

Reply:
OK, I was seeing "discovery engine" as more as the overall service ("engine") that provides block discovery, which includes advertising (which you factored out as advertiser) and lookup (which is still under "discovery engine"). As it is now, the client needs to be aware that you have lookups and advertisement as separate services, and then instantiate and start/stop them separately. Not what I'd do, but this rather subjective anyhow.

Reply:
Yeah the names don't line up nicely everywhere. We have:
discovery.nim which contains DiscoveryEngine type.
engine.nim which contains BlockExcEngine type.
this exists in a folder called engine, in a folder called blockexchange.
I've added:
advertiser.nim which contains Advertiser type.